src/lib/viewers/ResultsViewer.ts

```diff
@@ -0,0 +1,117 @@
+	public static readonly RESULTS_CUTOFF = 10;
+
+	public view(results: RunResults): void {
+		if (results.getViolationCount() === 0) {
+			this.display.displayInfo(getMessage(BundleName.ResultsViewer, 'summary.found-no-results'));
+		} else {
+			const violations = results.getViolations();
+			const violationCount = violations.length;
+			const fileCount = this.countUniqueFiles(violations);
+			// Output the total number of violations and files.
+			this.display.displayInfo(getMessage(BundleName.ResultsViewer, 'summary.found-results', [violationCount, fileCount]));
+
+			// Sort and output the violations.
+			const sortedViolations = this.sortViolations(violations);
+			const printableResultsCount = Math.min(ResultsDetailViewer.RESULTS_CUTOFF, sortedViolations.length);
+			const omittedResultsCount = sortedViolations.length - printableResultsCount;
+			for (let i = 0; i < printableResultsCount; i++) {
+				const violation = sortedViolations[i];
+				this.display.displayStyledHeader(getMessage(BundleName.ResultsViewer, 'summary.detail.violation-header', [i + 1, violation.getRule().getName()]));
+				const sev = violation.getRule().getSeverityLevel();
+				const primaryLocation = violation.getCodeLocations()[violation.getPrimaryLocationIndex()];
+				this.display.displayStyledObject({
+					engine: violation.getRule().getEngineName(),
+					severity: `${sev.valueOf()} (${SeverityLevel[sev]})`,
+					message: violation.getMessage(),
+					location: `${primaryLocation.getFile()}:${primaryLocation.getStartLine()}:${primaryLocation.getStartColumn()}`,
+					docs: violation.getResourceUrls().join(',')
+				}, ['engine', 'severity', 'message', 'location', 'docs']);
+			}
+
+			if (omittedResultsCount > 0) {
+				this.display.displayInfo(getMessage(BundleName.ResultsViewer, 'summary.detail.results-truncated', [omittedResultsCount]));
+			}
+
+			// Output the breakdown.
+			this.display.displayStyledHeader(getMessage(BundleName.ResultsViewer, 'summary.detail.breakdown.header'));
+			this.display.displayInfo(getMessage(BundleName.ResultsViewer, 'summary.detail.breakdown.total', [violations.length]));
+			for (const sev of Object.values(SeverityLevel)) {
+				if (typeof sev !== 'string') {
+					continue;
+				}
+				const sevCount = results.getViolationCountOfSeverity(SeverityLevel[sev] as SeverityLevel);
+				if (sevCount > 0) {
+					this.display.displayInfo(getMessage(BundleName.ResultsViewer, 'summary.detail.breakdown.item', [sevCount, sev]));
+				}
+			}
+		}
+	}
+
+	private countUniqueFiles(violations: Violation[]): number {
+		const fileSet: Set<string> = new Set();
+		violations.forEach(v => {
+			const primaryLocation = v.getCodeLocations()[v.getPrimaryLocationIndex()];
+			const file = primaryLocation.getFile();
+			if (typeof file === 'string') {
+				fileSet.add(file);
+			}
+		});
+		return fileSet.size;
+	}
+
+	private sortViolations(violations: Violation[]): Violation[] {
+		return violations.toSorted((v1, v2) => {
```

Comment:
`toSorted` is a recent addition, so I had to update the version of JS we're corresponding to. See the `tsconfig.json`.